### ISSUE TITLE
BitArray: add toggle range

### ISSUE THREAD

**Issue by asterite** [crystal-lang/crystal#2722]
Right now `BitArray` has a [toggle(index)](https://github.com/crystal-lang/crystal/blob/ea11f0ee1dd5f7999b3b72fb2fc74e6b5254cce9/src/bit_array.cr#L67) method to toggle an individual bit. It would be nice to have a method to toggle a range of bits, either with a range, from_index/to_index or start/count.

We need to discuss how the API should be (we can check how other languages do/provide this too) and then implement it.

**Comment by MaloJaffre** [comment]
I think it can accept a splat of ranges and then invert each range sequentially (to properly handle overlapping ranges):
```crystal
def toggle(*ranges)
  ranges.each do |range|
    toggle range
  end
end
def toggle(range)
  # Actual implementation
  ...
end
# Maybe:
def toggle(*, from, to)
  ...
end

def toggle(*, start, count)
  ...
end

**Comment by bcardiff** [comment]
> invert each range sequentially (to properly handle overlapping ranges):

I agree that is the cheapest implementation. But no sure if it is the proper thing to do. Overlapping ranges should be toggled only once  toggle 1..3, 2..5  should be the same as toggle 1..5.

But it would be to inefficient to track all that, so leaving it to the caller is fine.

**Comment by jhass** [comment]
I wouldn't add an overload for multiple ranges, let the caller handle that to me means to force them to iterate and call for each range explicitly or merge and call with one range.

**Comment by asterite** [comment]
Same comment as @bcardiff and @jhass, I would start with a simple method and we can later think about more complex stuff, though I'm not sure toggling multiple ranges at once is something common enough to be implemented in the standard library.
